text/0000-compact-slates.md

```diff
@@ -0,0 +1,499 @@
+| `part`        | secp256k1 AggSig     | (64) | If present                         |
+
+#### Optional Struct Status
+
+A bit field that denotes the presence or absence of the optional slate structures. Each bit is
+mapped to particular slate structure as follows:
+
+| Bit    | 7 | 6 | 5 | 4 | 3 | 2 | 1       | 0     |
+| -----: | - | - | - | - | - | - | ------- | ----- |
+| struct |   |   |   |   |   |   | `proof` | `coms`|
+
+If the corresponding field for a struct is 1, the struct is present and must be read accordingly.
+
+#### Coms Entries
+
+Coms Entries contains a length-prefixed array of entries corresponding to the `coms` struct.
+
+| Field   | type   | len    | notes                       |
+| ------: | ------ | ------ | --------------------------- |
+| length  | u16    | 2      |                             |
+| entries | struct | varies | array of entries, see below |
+
+Each Coms Entry is structured as follows:
+
+| Field       | type       | len  | notes                                                 |
+| ----------: | ---------- | ---- | ----------------------------------------------------- |
+| output flag | u8         | 1    | If non-zero, entry is output and contains `p` (proof) |
+| `f`         | u8         | 1    | features (1 = Coinbase, 0 = Plain)                    |
+| `c`         | Commitment | 33   |                                                       |
+| `p`         | Rangeproof | 675  | If present                                            |
+
+#### Proof
+
+Optional Payment proof, with fields as follows
+
+| Proof Field | type               | len  | notes                                |
+| ----------: | ------------------ | ---- | ------------------------------------ |
+| `saddr`     | ed25519 Public Key | 32   |                                      |
+| `raddr`     | ed25519 Public Key | 32   |                                      |
+| rsig flag   | u8                 | 1    | If non-zero, `rsig` field is present |
+| `rsig`      | ed25519 EDCSA Sig  | (64) | If present                           |
+
+#### Feature Args
+
+Optional feature args, presence or absence of which should be determined by the
+value of the `feat` field. Currently only present if `feat` is 1.
+
+|       Field | type  | len | notes                                |
+| ----------: | ----- | --  | ------------------------------------ |
+| `lock_hgt`  | u64   | 4   | Lock height, present if `feat` is 1  |
+|
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Is block header version needed?
+- Nerd Kernels need to be included
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+This RFC is envasaged as a necessary first step for all slate-exchange possibilities that would benefit from compactness, e.g:
+
+* [Slate Serialization](https://github.com/j01tz/grin-rfcs/blob/slate-serialization/text/0000-slate-serialization.md)
```

Comment:
Can replace slate-serialization and armored-slates here with [slatepack](https://github.com/mimblewimble/grin-rfcs/pull/55)